cdap-api/src/main/java/co/cask/cdap/api/schedule/Schedule.java

```diff
@@ -20,18 +20,30 @@
 /**
  * Defines a cron-based schedule for running a program. 
  */
-public class Schedule {
+public abstract class Schedule {
+  /**
+   * Schedule Type.
+   */
+  public enum ScheduleType { TIME, DATA }
```

Comment:
This info is serialized, and tells us what type we should use to deserialize the rest of the Schedule, either DataSchedule or TimeSchedule.